drivers/stepper/gpio_stepper_controller.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gpio_stepper
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/drivers/stepper.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/__assert.h>
+
+LOG_MODULE_REGISTER(gpio_stepper_motor_controller, CONFIG_STEPPER_LOG_LEVEL);
+
+#define NUMBER_OF_GPIO_PINS 4
+
+struct gpio_stepper_motor_controller_config {
+	const struct gpio_dt_spec *control_pins;
+};
+
+struct gpio_stepper_motor_controller_data {
+	const struct device *dev;
+	struct k_spinlock lock;
+	struct k_work_delayable stepper_dwork;
+	struct k_poll_signal *async_signal;
+	int32_t actual_position;
+	uint32_t delay_in_us;
+	int32_t step_count;
+};
+
+static int stepper_motor_move_with_4_gpios(const struct gpio_dt_spec *gpio_spec,
+					   int32_t step_number)
+{
+	switch (step_number) {
+	case 0:
+		LOG_DBG("Step 1010");
+		gpio_pin_configure_dt(&gpio_spec[0], GPIO_OUTPUT_ACTIVE);
```

Comment:
use the GPIO set function, not configuration – configuration is intent for pin behavior configuration, not level setup – on some arch, the set ist extremely faster than configuration and on really seldom arch the configuration is an complex interaction between different SoC function units